File: algorithms/sumPrimes.py
# The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

# Find the sum of all the primes below two million.

def sieve(n):
    nums = [True for i in range(n+1)]

    answer = 0    
    p = 2
    while (p*p <= n):
        if (nums[p] == True):
            for i in range(p*2, n+1, p):
                nums[i] = False
        p += 1

    for k in range(2, n):
        if nums[k]:
            answer += k    

    print(answer)


sieve(2000000)

File: algorithms/sumPrimes.js
// The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

// Find the sum of all the primes below two million.

let sieve = (n) =>{
    
    let myNums = new Array(n+1)
    for (let i=0; i<myNums.length; i++) {
        myNums[i] = true
    }

    console.log(myNums)

    for (let j=2; j*j<=n; j++) {
        if (myNums[j] == true) {
            for (let k = j*2; n+1; k=k*2) {
                myNums[k] = false
            }
        }
    }

    console.log(myNums)

    // let answer = 0
    // for (let l=2; l<=myNums.length; l++) {
    //     if (myNums[l] == true){
    //         answer += l
    //     }
    // }

    // return answer

}

console.log(sieve(200))